cmd/notary/tuf.go

```diff
@@ -269,56 +265,49 @@ func verify(cmd *cobra.Command, args []string) {
 	return
 }
 
-func passphraseRetriever() (string, error) {
-	fmt.Println("Please provide a passphrase for this root key: ")
-	var passphrase string
-	_, err := fmt.Scanln(&passphrase)
-	if err != nil {
-		return "", err
-	}
-	if len(passphrase) < 8 {
-		fmt.Println("Please use a password manager to generate and store a good random passphrase.")
-		return "", errors.New("Passphrase too short")
-	}
-	return passphrase, nil
-}
+func passphraseRetriever(keyName string, createNew bool, numAttempts int) (string, bool, error) {
+	fmt.Printf("Retrieving passphrase for key %s: ", keyName)
 
-func getPassphrase(confirm bool) ([]byte, error) {
-	if pass := os.Getenv("NOTARY_ROOT_PASSPHRASE"); pass != "" {
-		return []byte(pass), nil
+	if (numAttempts > 3 && !createNew) {
+		return "", true, errors.New("Too many attempts")
```

Comment:
I think the retry logic should be inside the password verifier, not passphraseRetriever. It would be easy for a call site that needs to retrieve a passphrase to forget to retry in a loop if the password is incorrect. It's also a little weird that the retry limit is enforced in passphraseRetriever when it's not the function actually doing the retries.